wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -1681,7 +1681,61 @@ public final class MLSService: MLSServiceInterface {
     // MARK: - Proteus to MLS Migration
 
     public func startProteusToMLSMigration() async throws {
-        // Migrate proteus team group conversations to MLS
+        guard let context = context else {
+            assertionFailure("MLSService.context is nil")
+            return
+        }
+
+        let groupConversations = try await context.perform {
+            try ZMConversation.fetchAllTeamGroupConversations(messageProtocol: .proteus, in: context)
+                .map {(
+                    $0.qualifiedID,
+                    $0.mlsGroupID,
+                    $0.localParticipants.map { MLSUser(from: $0) }
+                )}
+        }
+        for (qualifiedID, mlsGroupID, members) in groupConversations {
+
+            guard let qualifiedID else {
+                assertionFailure("the group conversation has no `qualifiedID` set")
+                continue
+            }
+            guard let mlsGroupID else {
+                assertionFailure("the group conversation has no `mlsGroupID` set")
+                continue
+            }
+
+            do {
+
+                // update message protocol to `mixed`
+                try await actionsProvider.updateConversationProtocol(qualifiedID: qualifiedID, messageProtocol: .mixed, context: context.notificationContext)
+
+                // sync the group conversation
+                try await actionsProvider.syncConversation(qualifiedID: qualifiedID, context: context.notificationContext)
+
```

Comment:
why do we pass context.notficationContext?

Reply:
it's needed by the actions to send a notification about requests being available